projects/demo-playwright/tests/kit/input-number/input-number.spec.ts

```diff
@@ -0,0 +1,336 @@
+    tuiGoto,
+} from '@demo-playwright/utils';
+import {expect, Locator, test} from '@playwright/test';
+
+test.describe('InputNumber', () => {
+    let example: Locator;
+    let input: Locator;
+
+    test.describe('API', () => {
+        test.beforeEach(({page}) => {
+            example = new TuiDocumentationApiPagePO(page).apiPageExample;
+            input = example.locator(
+                '[automation-id=tui-primitive-textfield__native-input]',
+            );
+        });
+
+        test('Infinite precision', async ({page}) => {
+            await tuiGoto(page, '/components/input-number/API?precision=Infinity');
+            await input.focus();
+            await input.fill('1,2345');
+            await expect(example).toHaveScreenshot('01-input-number.png');
+        });
+
+        test('prefix + value + postfix', async ({page}) => {
+            await tuiGoto(page, '/components/input-number/API?prefix=$&postfix=GBP');
+            await expect(example).toHaveScreenshot('02-input-number.png');
+            await input.clear();
+            await expect(example).toHaveScreenshot('03-input-number.png');
+        });
+
+        ['left', 'right'].forEach((align, i) => {
+            test(`align=${align}`, async ({page}) => {
+                const readableFormatText = 'Very long text';
+
+                await tuiGoto(
+                    page,
+                    `/components/input-number/API?style.text-align=${align}&prefix=${readableFormatText}&postfix=${readableFormatText}`,
+                );
+                await expect(example).toHaveScreenshot(`04-input-number-${i}.png`);
+            });
+        });
+
+        test.describe('overflow', () => {
+            ['75', '140', '158'].forEach((sandboxWidth, i) => {
+                test(`sandboxWidth=${sandboxWidth}`, async ({page}) => {
+                    await tuiGoto(
+                        page,
+                        `/components/input-number/API?tuiTextfieldPostfix=$&tuiTextfieldPrefix=VeryLongText&sandboxWidth=${sandboxWidth}`,
+                    );
+
+                    await expect(example).toHaveScreenshot(`05-input-number-${i}.png`);
+                });
+            });
+        });
+    });
+
+    test.describe('Examples', () => {
+        test('cursor position', async ({page}) => {
+            await tuiGoto(page, '/components/input-number');
+
+            const example = new TuiDocumentationPagePO(page).getExample('#currency');
+
+            await expect(example).toHaveScreenshot('06-input-number.png');
+
+            const inputs = await example
+                .locator('[automation-id="tui-primitive-textfield__native-input"]')
+                .all();
+
+            for (const [i, input] of inputs.entries()) {
+                await input.clear();
+                await input.focus();
+                await expect(example).toHaveScreenshot(`06-input-number-${i}.png`);
+            }
+        });
+    });
+
+    test.describe('Caret navigation', () => {
```

Comment:
The same https://github.com/taiga-family/taiga-ui/pull/6441#discussion_r1450413595

Reply:
fixed